### ISSUE TITLE
await app.render does not seem to be blocking

### ISSUE THREAD

**Issue by serle** [zeit/next.js#2267]
Koa handler:

await app.render(ctx.req, ctx.res, ctx.path, ctx.query);
ctx.respond = false

browser gets not found response, while next builds: 

Building page: /app
 DONE  Compiled successfully in 2513ms00:56:35
 WAIT  Compiling...00:56:36
 DONE  Compiled successfully in 214ms00:56:36

Any suggestions?

**Comment by arunoda** [comment]
Interesting, can I have a sample app for this?

**Comment by serle** [comment]
Ok I have found the problem after some debugging inside the next code. We need to set 
`ctx.respond = false`  BEFORE the `await next.render(...)` so that next does not think that the response is already done and not render the component. The Koa sample application seems to be wrong in this regard and needs to be updated. I would prefer the 'owners' of the sample to comment.

**Comment by arunoda** [comment]
Feel free to send us a PR.

**Comment by serle** [comment]
The work around fix is to set `ctx.respond = false` before the `next.render()` call, but if one has passed the render responsibility to next, then is it necessary for it to test if the response == false and subsequently not render. Any opinion on the matter?

**Comment by arunoda** [comment]
I assume await doesn't work well inside koa. May be you can convert the the promise return by app.render() into a thunk and yeild it.

**Comment by serle** [comment]
await works perfectly, in fact that is way I went with Koa as opposed to express, because the whole system uses promises and generators. As I said above the problem is in the next code where it evaluates if `res.respond == false` and then returns prematurely, by resolving the promise without actually producing a response. I don't understand why this check is necessary give that I have handed off the rendering to next.
